server/lib/api/users.js

```diff
@@ -97,3 +106,33 @@ async function resetPassword(ctx, next) {
     ctx.status = 204
   })
 }
+
+async function verifyEmail(ctx, next) {
+  const { id } = ctx.params
+  const { code } = ctx.query
+  const { email } = ctx.request.body
+
+  if (!id || !code || !isValidEmail(email)) {
+    throw new httpErrors.BadRequest('Invalid parameters')
+  }
+
+  const verifiedEmail = await useEmailVerificationCode(id, email, code)
+  if (!verifiedEmail) {
+    throw new httpErrors.BadRequest('Email verification code is invalid')
```

Comment:
When you implement this on the client, you'll probably want to mention that the code might be expired.